site/resources/views/livewire/finder-js.blade.php

```diff
@@ -0,0 +1,236 @@
+                element => {
+                    const key = element.getAttribute("data-key");
+                    if (!this.openedFolder.includes(key)) {
+                        this.openedFolder.push(key);
+                    }
+                },
+            );
+        },
+
+        /**
+         * Expand (open) or collapse (close) the given folder.
+         */
+        toggleOpen(element, key) {
+            this.openedFolder = _.xor(this.openedFolder, [key]);
+
+            // Unselect children if folder is closed and not selected.
+            if (!this.selectedItems.includes(key)) {
+                _.pull(
+                    this.selectedItems,
+                    ..._.map(
+                        element.closest('li').querySelectorAll('li'),
+                        childs => childs.getAttribute('data-key'),
+                    ),
+                );
+            }
+
+            this.closeAllDropDowns(element);
+        },
+
+        /**
+         * Open the contextual menu for the given item.
+         */
+        openMenu(element, keepSelection = false) {
+            // Close all dropdowns except the one clicked.
+            // This is needed to be done manually because some
+            // events are not bubbled (preventPropagation).
+            this.closeAllDropDowns(element);
+
+            // Unselect items to avoid confusion on the targeted
+            // menu actions.
+            if (!keepSelection) this.selectedItems = [];
+        },
+
+        /**
+         * Close all open contextual menu.
+         */
+        closeAllDropDowns(element = null) {
+            document.querySelectorAll(
+                '[data-bs-toggle="dropdown"]',
+            ).forEach((dropdown) => {
+                if (dropdown !== element) {
+                    bootstrap.Dropdown.getInstance(dropdown)?.hide();
+                }
+            });
+        },
+
+        /**
+         * Select all items.
+         */
+        selectAll() {
+            this.selectedItems = _.map(
+                document.querySelectorAll('.finder-folder, .finder-card'),
+                children => children.getAttribute('data-key'),
+            );
+        },
+
+        /**
+         * Return if all items are selected.
+         */
+        isAllSelected() {
+            return this.selectedItems.length === document.querySelectorAll('.finder-folder, .finder-card').length;
+        },
+
+        /**
+         * Open a prompt and aks the user to change the name of the
+         * given folder.
+         */
+        renameFolder($wire, folderId, reloadAfterSave = false) {
+            const newName = prompt("{{ trans('courses.finder.menu.rename_prompt') }}");
+            if (newName !== null) {
+                $wire.call("renameFolder", folderId, newName, reloadAfterSave);
+            }
+        },
+
+        /**
+         * Initialize the Sortable plugin.
+         */
+        initSortable(list) {
```

Comment:
Move sortable initialization to [sortable.js](https://github.com/unil-lettres/impact/blob/development/site/resources/js/sortable.js) file ?

Reply:
It must be initialized dynamically for every new folder created and sortable.js is only called when the page is loaded.